search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java

```diff
@@ -161,7 +140,54 @@ private OpenSearchSwsResponse handleResponse(HttpResponse<String> response) {
             throw new RuntimeException(response.body());
         }
         return attempt(() -> singleLineObjectMapper.readValue(response.body(), OpenSearchSwsResponse.class))
-            .orElseThrow();
+                   .orElseThrow();
+    }
+
+    private AbstractQueryBuilder<?> multiMatchQuery(ResourceAwsQuery query) {
+        var field = query.getValue(FIELDS).toString();
+        return QueryBuilders
+                   .multiMatchQuery(query.getValue(SEARCH_ALL).toString(), extractFields(field))
+                   .operator(Operator.AND);
+    }
+
+    private BoolQueryBuilder boolQuery(ResourceAwsQuery query) {
+        var bq = QueryBuilders.boolQuery();
+        query.getLuceneParameters()
+            .forEach((key, value) -> {
+                var swsKey = key.swsKey().toArray(String[]::new);
+                if (swsKey.length > SINGLE_FIELD) {
+                    bq.must(QueryBuilders.multiMatchQuery(value, swsKey));
+                } else {
+                    if (key.operator() != ParameterKey.Operator.EQUALS) {
+                        bq.must(rangeQuery(key.operator(), swsKey[0], value));
```

Comment:
Can you explain these new changes to boolQuery??

Reply:
A lot of logic here is tightly coupled to the search engine, and is hard for me to understand when i dont know how search works. 

Maybe consider adding some javadoc for readability... Not a must have, but makes it easier to understand